### PR TITLE
DPL: use Framework/RuntimeError helpers throughout the framework

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Review by aphecetche — DISMISSED** [review]

File: Detectors/TPC/workflow/include/TPCWorkflow/CATrackerSpec.h [hunk 1]
```diff
@@ -140,6 +140,5 @@ struct Config {
 /// @param tpcsectors list of sector numbers
 framework::DataProcessorSpec getCATrackerSpec(o2::tpc::reco_workflow::CompletionPolicyData* policyData, ca::Config const& specconfig, std::vector<int> const& tpcsectors);
 
-o2::framework::CompletionPolicy getCATrackerCompletionPolicy();
```
Comment (davidrohr) [inline Detectors/TPC/workflow/include/TPCWorkflow/CATrackerSpec.h]: Actually I don't understand what this does or why it was here. Is it safe to remove it?

**Comment by davidrohr** [comment]
In general, it is not clear to me what is the difference between std::runtime_error and the framework version?

File: Detectors/TPC/workflow/include/TPCWorkflow/CATrackerSpec.h (hunk 1 above)
Comment (ktf) [reply Detectors/TPC/workflow/include/TPCWorkflow/CATrackerSpec.h]: Good catch, I dropped it to debug something else.

**Comment by ktf** [comment]
Among other things:

* It (optionally) allows to capture the backtrace of where the exception was thrown. 
* It avoids including `<string>`, `<stdexecept>`, `<fmt.h>`, all of which  are compiletime heavyweights, when you want to throw a string based exception.
* It does not allocate on throw.

**Comment by davidrohr** [comment]
ok, looks ok to me except for the line with the completion policy, which is from Matthias and I have no idea what it does :)

**Comment by ktf** [comment]
I will put it back.

**Pull request merged**